### ISSUE TITLE
Choosing the serializer when signing messages

### ISSUE THREAD

**Issue by xavierhardy** [celery/celery#3954]
http://docs.celeryproject.org/en/latest/userguide/security.html#message-signing

"To enable this you should configure the `task_serializer` setting to use the `auth` serializer."

Which serializer is used alongside the message signature mechanism? How to change it? Let's say I want to use `yaml` with message signing, is it possible?

**Comment by georgepsarakis** [comment]
`msgpack`, `pickle` and `yaml` are considered insecure serialization formats. 

If you notice [here](https://github.com/celery/celery/blob/master/celery/security/serialization.py#L96), the `auth` serializer is registered and is using the JSON serializer to encode signed messages.

I think that you could also create your own serializer, but I do not see any benefit.

**Comment by xavierhardy** [comment]
OK, so you're saying that this line means that you are exclusively using JSON.
```
Note that this will also disable all insecure serializers so that the worker won’t accept messages with untrusted content types.
```
I think that should be clearly stated. I'm fine with the JSON, as I agree about the other formats being insecure and experimental.
Should I set all of the serializers to `auth` and the `accept_content` option to `["auth"]` then? Are there any requirements on the permissions of the certificates and key files?

**Comment by georgepsarakis** [comment]
Yes, setting the serializers to `auth` should enable message signing. Regarding the SSL file permissions, you should look into the requirements of [OpenSSL.crypto](https://github.com/celery/celery/blob/master/celery/security/utils.py#L9).

**Comment by georgepsarakis** [comment]
@xavierhardy do you perhaps think the documentation needs to be expanded? Please feel free to open a Pull Request, if you would like.
